core/src/main/java/org/ehcache/core/Ehcache.java

```diff
@@ -217,44 +162,29 @@ public V apply(final K k) {
   }
 
   @Override
-  public void put(final K key, final V value) throws CacheWritingException {
+  public void put(final K key, final V value) {
     putObserver.begin();
     statusTransitioner.checkAvailable();
     checkNonNull(key, value);
-    final BiFunction<K, V, V> remappingFunction = memoize(new BiFunction<K, V, V>() {
-      @Override
-      public V apply(final K key, final V previousValue) {
-        try {
-          if (cacheLoaderWriter != null) {
-            cacheLoaderWriter.write(key, value);
-          }
-        } catch (Exception e) {
-          throw new CachePassThroughException(newCacheWritingException(e));
-        }
-        return value;
-      }
-    });
 
     try {
-      ValueHolder<V> computed = store.compute(key, remappingFunction);
-      if (computed != null) {
-        putObserver.end(PutOutcome.ADDED);
-      } else {
+      PutStatus status = store.put(key, value);
+      switch (status) {
+      case PUT:
+        putObserver.end(PutOutcome.PUT);
```

Comment:
I wouldn't do that as this would bastardize the statistics types and would make M&M work slightly harder.

If you like this idea, think about doing it the other way: returning a `PutStatus` from put but IMHO this would couple the `Store` contract with the statistics one. I personally prefer this switch statement.